### PR TITLE
save a fullscreen state when Kiosk mode is called

### PR DESCRIPTION
**Pull request by watilde**
Hello!

This patch fixes https://github.com/electron/electron/issues/8338, and I've added some of the test cases related to the patch.

Here is an example code:
+ https://gist.github.com/watilde/c47eba263e6048c93ebf7c90a5e60c00

### CODE REVIEW

File: atom/browser/native_window_mac.h [hunk 1]
```diff
@@ -162,6 +162,8 @@ class NativeWindowMac : public NativeWindow,
 
   bool is_kiosk_;
 
+  bool was_fullscreen_;
```
Comment (kevinsawicki) [inline atom/browser/native_window_mac.h]: This should be initialized to `false` in https://github.com/watilde/electron/blob/9bdaf30d22d93f4c9c294ce114f0103b4073b47b/atom/browser/native_window_mac.mm#L621-L622

Comment (watilde) [inline atom/browser/native_window_mac.h]: Oh, I missed it! Thanks for your comment, I've updated.

File: spec/api-browser-window-spec.js [hunk 2]
```diff
@@ -1417,6 +1417,42 @@ describe('BrowserWindow module', function () {
...
+
+    describe('fullscreen state', function () {
+      // Only implemented on macOS.
+      if (process.platform !== 'darwin') return
+
+      it('can be changed with setFullScreen method', function () {
+        w.destroy()
+        w = new BrowserWindow()
+        w.setFullScreen(true)
+        assert.equal(w.isFullscreen(), true)
```
Comment (kevinsawicki) [inline spec/api-browser-window-spec.js]: I think this needs to be `w.isFullScreen()` with a capitalized `S`. Also a few other cases of this below as well.

Comment (watilde) [inline spec/api-browser-window-spec.js]: That's right! I've updated.

**Comment by kevinsawicki** [comment]
Thanks for fixing this 👍 🚢

**Pull request merged**